src/GDGUkraine/utils.py

```diff
@@ -114,3 +122,283 @@ def make_vcard(user_reg, url=None):
+            else:
+                rkwargs[param.name] = ikwargs.pop(param.name)
+        elif param.kind == inspect.Parameter.VAR_KEYWORD:
+            # param is **kwargs
+            rkwargs.update(ikwargs)
+            ikwargs.clear()
+
+    # Check whether there's any params exceeding ones
+    # declared into original handler
+    if len(iargs):
+        raise TypeError('Too many positional arguments passed!')
+    elif len(ikwargs):
+        raise TypeError('Too many keyword arguments passed!')
+
+    # Build URI path string ending for concatenation with base URI
+    uargs = '/'.join([urllib.parse.quote_plus(_)
+                      for _ in rargs if _])
+    # Build GET params list string
+    ukwargs = '&'.join(['='.join([urllib.parse.quote_plus(k),
+                                  urllib.parse.quote_plus(str(v))])
+                        for k, v in rkwargs.items() if v])
+
+    if uargs:
+        url = '/'.join([url, uargs])
+
+    if ukwargs:
+        url = '?'.join([url, ukwargs])
+
+    # Return final URI string
+    return url
+
+
+def build_url_map(force=False):
+    '''Builds resolve map for class-based routes
+        build_url_map(force=True) is called by url map builder cherrypy plugin
+    '''
+
+    def retrieve_class_routes(cls, mp, handler_cls=None):
+        if handler_cls is None:
+            handler_cls = '.'.join([cls.__class__.__module__,
+                                    cls.__class__.__name__]).lower()
+        if not mp.endswith('/'):
+            mp = '/'.join([mp, ''])
+        res = {}
+        for method in dir(cls):
+            hndlr = getattr(cls, method)
+            uri = mp
+            if hasattr(hndlr, '__name__') \
+                    and hndlr.__name__ != 'index' \
+                    and method:
+                uri = ''.join([mp, method])
+            if not uri:
+                uri = '/'
+
+            if inspect.ismethod(hndlr):
+                if getattr(hndlr, 'exposed', False):
+                    if hndlr.__name__ != method:
+                        continue
+                    # That's it! It's a final method
+
+                    # (args_, varargs_, varkw_, values_) = \
+                    #     inspect.getargspec(hndlr)
+                    # pprint(inspect.getargspec(hndlr))
+                    # Use inspect.getargspec
+                    # instead of inspect.signature for Python < 3.5
+                    params = inspect.signature(hndlr).parameters
+
+                    key_cls = handler_cls
+                    if hndlr.__name__ != 'index':
+                        key_cls = '.'.join([handler_cls, hndlr.__name__])
+
+                    if res.get(key_cls):
+                        continue
+
+                    res[key_cls] = {
+                        'args': params,
+                        'url': (uri
+                                if uri and uri.endswith('/')
+                                else '/'.join([uri]))}
```

Comment:
'/'.join([uri]) is the same as uri
```python
>>> uri = 'asd'
>>> '/'.join([uri]) == uri
True
```

I suppose that should be '/'.join([uri,'']), like at line 213.